### ISSUE TITLE
package_to_hub failing on windows

### ISSUE THREAD

**Issue by MattStammers** [huggingface/huggingface_sb3#36]
Dear Thomas,

Firstly, I wanted to thank you for your excellent tutorials on deep learning. I have found them fantastically interesting and enjoyable. There is a glitch with vizdoom preventing me from installing this on google collab but that is not your problem to address - everything else so far has worked fine apart from this package_to_hub() call.

When running in Google collab it works fine. However, every time I try it on a windows machine, even as per your example:

```python
import gymnasium as gym

from stable_baselines3 import PPO
from stable_baselines3.common.env_util import make_vec_env
from huggingface_sb3 import package_to_hub

# Create the environment
env_id = "LunarLander-v2"
env = make_vec_env(env_id, n_envs=1)

# Create the evaluation env
eval_env = make_vec_env(env_id, n_envs=1)

# Instantiate the agent
model = PPO("MlpPolicy", env, verbose=1)

# Train the agent
model.learn(total_timesteps=int(5000))

# This method save, evaluate, generate a model card and record a replay video of your agent before pushing the repo to the hub
package_to_hub(model=model, 
               model_name="ppo-LunarLander-v2",
               model_architecture="PPO",
               env_id=env_id,
               eval_env=eval_env,
               repo_id="ThomasSimonini/ppo-LunarLander-v2",
               commit_message="Test commit")
```

I first get an error: ModuleNotFoundError: No module named 'gym' because it is looking for gym but the package is called gymnasium. Installing openAI's gym causes the import to succeed but this should not be necessary. Later when I call  package_to_hub() even if I don't import openAI's gym I cannot push to the hub. 

I wish I could explain why this is happening but I have tried it on many environments now and it never happens in Linux but always happens in Windows environments even if they are otherwise ostensibly identical. push_to_hub() works but package_to_hub() does not. 

I could probably force-fix it by installing gymnasium as gym but this is going to confuse newcomers. If you could fix this in the next version would be greatly appreciated as I do rather like the videos being uploaded and as a result am burning through a lot of collab credits 😂.

Hope that makes sense.

yours sincerely,

Matt

**Comment by MattStammers** [comment]
Also. When you do get it to install (which I have been able to a couple of times. You get the following error when you use the package_to_hub() command:

```python
     98 def is_atari(env_id: str) -> bool:
     99     """
    100     Check if the environment is an Atari one
    101     (Taken from RL-Baselines3-zoo)
    102     :param env_id: name of the environment
    103     """
--> 104     entry_point = gym.envs.registry.env_specs[env_id].entry_point
    105     return "AtariEnv" in str(entry_point)

AttributeError: 'dict' object has no attribute 'env_specs'
```

**Comment by MattStammers** [comment]
I think I have worked out why this is happening by accident.

The discrepancy seems to be occurring because gym==0.25.1 and numpy=1.23.1 is loaded into the collab environment and numpy==1.24.1 gym==0.26.2 is loaded into the Windows environment. 

By patching the numpy version alone and bumping it to 1.24.1 (but leaving the gym environments as they are) the models are still working but the package_to_hub() function is now working. Thus I assume the issue is caused by the numpy minor version going from 1.23 to 1.24. 

Future releases should therefore probably fix the numpy version at 1.23.1 for now until a more permanent fix can have dedicated development resources allocated to it.

**Comment by MattStammers** [comment]
The above fix works if you train the environment on one machine and then port it to another to upload but doesn't work if it is all on the same machine. To make that work you also need to roll gym back to 0.25.1 or 0.25.2 then it works (but in my case, the video upload goes wrong and it reverts to a push_to_hub. Definitely, more work to do here ;D but we look forward to the next release
